nixos/default.nix

```diff
@@ -26,66 +40,115 @@ in {
             # fontconfig by default.
             fonts.fontconfig.enable = lib.mkDefault
               (cfg.useUserPackages && config.fonts.fontconfig.enable);
-
             # Inherit glibcLocales setting from NixOS.
             i18n.glibcLocales = lib.mkDefault config.i18n.glibcLocales;
+
+            # .ssh/config needs to exists before login to let ssh login as that user
+            programs.ssh.internallyManaged =
+              lib.mkDefault (!cfg.useUserService);
           };
         }];
+
+        systemd.services = mapAttrs' (_:
+          { home, programs, ... }:
+          let inherit (home) username homeDirectory;
+          in nameValuePair "ssh_config-${utils.escapeSystemdPath username}" {
+            enable = with programs.ssh; enable && !internallyManaged;
+            description = "Linking ${username}' ssh config";
+            wantedBy = [ "multi-user.target" ];
+            before = [ "systemd-user-sessions.service" ];
+
+            unitConfig.RequiresMountsFor = homeDirectory;
+            stopIfChanged = false;
+            serviceConfig = (baseService username) // {
+              User = username;
+              ExecStart = [
+                "${pkgs.coreutils}/bin/mkdir -p ${homeDirectory}/.ssh"
+                "${pkgs.coreutils}/bin/ln -s ${programs.ssh.configPath} ${homeDirectory}/.ssh/config"
+              ];
+            };
+          }) cfg.users;
```

Comment:
This fails with `error: The option `home-manager.systemd' does not exist.` which makes the entire thing basically unuseable. How did you even test this PR if it fails to eval?

Reply:
You probably meant to put that definition outside of the `home-manager` set, it seems to work as intended if doing so.